iroha-lib/model/Query.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef QUERY_HPP
+#define QUERY_HPP
+
+#include "LibsAndClassDeclarations.h"
+
+#include "crypto/keypair.hpp"
+#include "queries.pb.h"
+#include "generators/QueryGenerator.hpp"
+
+
+namespace iroha_lib {
+
+class Query {
+
+private:
+    uint64_t counter_;
+    uint64_t created_time_;
+    iroha::protocol::Query protobuf_query_;
+    iroha::keypair_t keypair_;
+    QueryGenerator query_generator_;
+
+public:
+    Query(
+            const iroha::keypair_t& keypair,
+            uint64_t counter = 0u,
+            const uint64_t created_time = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count());
+
+    Query& getAccount(const std::string& account_id);
+    Query& getAccountAssets(const std::string& account_id);
+    Query& getAccountDetail(const std::string& account_id);
+    Query& getAccountTransactions(const std::string& account_id);
+    Query& getAccountAssetTransactions(
+            const std::string& account_id,
+            const std::string& asset_id);
+    Query& getTransactions(
+            const std::string& account_id,
+            const std::vector<std::string>& tx_hashes);
+    Query& getSignatories(const std::string& account_id);
+    Query& getAssetInfo(
+            const std::string& account_id,
+            const std::string& asset_id);
+    Query& getRoles(const std::string& account_id);
+    Query& getRolePermissions(
+            const std::string& account_id,
+            const std::string& role_id);
```

Comment:
Given these queries are often going to be used as constructor methods, we could set these to `Query&&` and make things easier for the compiler.

Reply:
Those methods are returning `*this;`, so we can create query like that:
```
    return generateSampleQuery(
                account_name,
                key_path,
                counter)
            .getAccountAssets(account_name)
            .signAndAddSignature();
```
So Maybe returning as `Query&` is not that bad, because we are changing the same object?

Reply:
Yes, you're right. It should still be possible to return an r-value reference via `std::move`, but it might actually pessimise performance.